### DISCUSSION TITLE
Where is builder in nuxt3 (it was in @nuxt/builder previously)?

### DISCUSSION THREAD

**Discussion by hnviradiya** [nuxt/nuxt#16606] [Questions] [answered]
Nuxt 2 was having builder.

Is that not part of nuxt 3?

import { Builder } from '@nuxt/builder';

**Comment by danielroe** [comment, marked as answer]
There is `@nuxt/vite-builder` and `@nuxt/webpack-builder`

**Reply by hnviradiya** [reply]
Thanks @danielroe I was able to run that. Now when I run it. It is expecting css folder and giving below error where as there is no such folder generated by nuxt build.

`[vite:load-fallback] Could not load [c:/Users/hardik/Documents/GitLab/project/src/client-app/.nuxt/css]() (imported by node_modules/nuxt3/dist/app/entry.mjs): ENOENT: no such file or directory, open '[c:\Users\hardik\Documents\GitLab\project\src\client-app\.nuxt\css]()'`

I have used following to run nuxt programmatically.

```
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module.js';
import { loadNuxt } from 'nuxt3';
import { bundle } from '@nuxt/vite-builder-edge';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);

  // Check if we need to run Nuxt in development mode
  const isDev = process.env.NODE_ENV !== 'production'

  // Get a ready to use Nuxt instance
  const nuxt = await loadNuxt({ rootDir: 'src/client-app/' })

  // Enable live build & reloading on dev
  if (isDev) {
    bundle(nuxt)
  }

  await app.listen(3001);
}
bootstrap();
```

**Reply by danielroe** [reply]
Please see my comment in your issue.

**Reply by hnviradiya** [reply]
Thanks a lot man. I was having difficult time with documentation and seems I am using this too early. Probably this will be last thing I will need from you.

In v2 we were having `nuxt.render(req, res)` or `nuxt.renderRoute(route, context)`. What can we use equivalently in nuxt 3? I am not seeing these methods in v3.

https://nuxtjs.org/docs/internals-glossary/nuxt/

**Reply by danielroe** [reply]
No, the server is implemented differently. You can see the internal implementation in nuxt/test-utils:

https://github.com/nuxt/framework/blob/39f7eb2cd21d6e3a5642eca877d5f4fc6d28f840/packages/test-utils/src/server.ts#L1-L37

**Reply by hnviradiya** [reply]
This seems to be running server separately in entirely separate process. `nuxt.render(req, res)` or `nuxt.renderRoute(route, context)` are working as middleware.

**Reply by hnviradiya** [reply]
I have opened separate discussion for render if there is something have solution to. From nitro, I think [addDevServerHandler ](https://github.com/nuxt/framework/blob/2cf9009d33052c061be2ddf4227803528d2d3a4c/packages/kit/src/nitro.ts#L46) can be used, but I am unable to find its test case.

https://github.com/nuxt/framework/discussions/8233

**Reply by hnviradiya** [reply]
Because in server.ts also it is running in seperate thread with different ports, which entirely not fitting in the original requirement of serving nuxt from same web app as backend api.

**Comment by hnviradiya** [comment]
Link to answer to using builder.

https://github.com/nuxt/framework/issues/3699
